app/utils/graph-tools.js

```diff
@@ -63,7 +63,7 @@ const findJob = (jobs, jobId) => jobs.find(j => j && `${j.id}` === `${jobId}`);
  * @return {Object}         Reference to the job object from the list if found
  */
 const prJob = (jobs, prNum, name) =>
-  jobs.find(j => j && j.group === prNum && j.name.endsWith(name));
+  jobs.find(j => j && j.group === prNum && j.name.endsWith(`:${name}`));
```

Comment:
```suggestion
  jobs.find(j => j && j.group === prNum && j.name.split(`PR-${prNum}`)[1] === `:${name}`);
```
This handles the situation where you have a PR job that is wrapped in a stage:
```
jobs:
  p1:
    requires: ~pr
  foo:
    requires: ~pr
    steps:
        - build: echo FOO
  bar-foo:
    requires: ~pr
    steps:
        - build: echo BAR-FOO
  setup:
    requires: ~pr
    steps:
        - build: echo SETUP
  teardown:
    requires: ~pr
    steps:
        - build: echo TEARDOWN
  s1:
    requires: []
    steps:
        - build: echo S1

  build:
    requires: ~commit

  b2:
    requires: build

stages:
    puff:
        jobs: [s1]
        requires: [~pr]
        setup:
            steps:
              - build: echo STAGE SETUP
        teardown:
            steps:
              - build: echo STAGE TEARDOWN
```